### ISSUE TITLE
Storing the model whit best accuracy on the evaluation data.

### ISSUE THREAD

**Issue by alejandrojcastaneira** [explosion/spaCy#3706]
Hello, I'm trying to use a training script including evaluation on a validation set, so I would like to keep the track of the best f1-score achieved after every iteration, and later save only the best state as the final model, so theoretically this could prevent from over-fitting.

after every iteration:

 ```
with nlp.use_params(optimizer.averages):

     results = evaluate()

     if (results["ents_f"]) >= current_set_best_results:

          best_ner = nlp.get_pipe('ner')

          current_set_best_results = results["ents_f"]
```
so later after the training:
```
   nlp.replace_pipe('ner', best_ner)

   nlp.to_disk("/model")
```
My question is if I'm using int the correct way the nlp.use_params(), nlp.get_pipe() and nlp.replace_pipe() functions?

Best Regards.

## Which page or section is this issue related to?

https://spacy.io/usage/training#tips

**Comment by honnibal** [comment]
That does seem correct. For simplicity, what I often do is create the final model package by copying in the different directories with the best NER model, best tagger model, best parser model etc. You need to take care to make sure your final model has all the relevant strings for the labels, but otherwise this approach should work fine --- the component models are independent, so you can just copy the directories around.

**Comment by alejandrojcastaneira** [comment]
Thanks you, for letting me know, I also thougth on saving the model after each iteración similar to the CLI behaviour, but then I had the other pipelines disabled, so those are not saved to disk, so perhaps it could be useful to have an enable_pipe function? best regards
